nemo/utils/sequence_packing_utils.py

```diff
@@ -212,21 +220,31 @@ def fill_packing_strategy(
         per_seq_data = sequences[seq_len]
         if len(per_seq_data) > 0:
             perm = np.random.permutation(len(per_seq_data))
-            input_ids = np.array([x['input_ids'] for x in per_seq_data])[perm].tolist()
+            input_ids = np.array([x["input_ids"] for x in per_seq_data])[perm].tolist()
             try:
-                loss_mask = np.array(
-                    [
-                        [
-                            # (x['answer_start_idx'] - 1) because we want to train on the output
-                            # after the last context token
-                            idx >= (x['answer_start_idx'] - 1) and x['input_ids'][idx] != pad_id
-                            for idx in range(len(x['input_ids']))
-                        ]
-                        for x in per_seq_data
-                    ]
-                )[perm].tolist()
+                loss_mask = np.array([x["mask"] for x in per_seq_data])[perm].tolist()
```

Comment:
Just to make sure, is the "mask" key only for chat dataset? If so could you add a comment here to explain

Reply:
Good call out, will ensure it is loss_mask for all modalities, updating now.